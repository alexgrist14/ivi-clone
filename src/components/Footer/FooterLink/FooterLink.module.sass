.link
  display: block
  color: rgba(255,255,255,.48)
  text-decoration: none
  transition-duration: .4s
  transition-property: color,box-shadow,opacity

  &:hover, &:focus-visible
    color: white
