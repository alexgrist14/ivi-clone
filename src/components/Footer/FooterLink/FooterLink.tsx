import Link from "next/link";
import { FC } from "react";
import {useTranslation} from "react-i18next";
import styles from "./FooterLink.module.sass";
import ILink from "/src/types/ILink";

const FooterLink: FC<ILink> = ({ content, url, target }) => {
  const { t } = useTranslation();

  return (
    <Link target={target} className={styles.link} href={url}>
      {t(content as string)}
    </Link>
  );
};

export default FooterLink;
