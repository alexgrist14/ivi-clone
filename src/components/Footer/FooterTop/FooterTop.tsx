import { FC } from "react";
import styles from "./FooterTop.module.sass";
import FooterList from "../FooterList/FooterList";
import FooterButtons from "../FooterButtons/FooterButtons";
import FooterAsk from "../FooterAsk/FooterAsk";
import FooterWidget from "../FooterWidget/FooterWidget";
import { topLeftLinks, topRightLinks } from "./FooterTop.utils";
import FooterCert from "../FooterCert/FooterCert";

const FooterTop: FC = () => {
  return (
    <div className={styles.top}>
      <div>
        <span className={styles.top__title}>О нас</span>
        <nav className={styles.top__nav}>
          <FooterList items={topLeftLinks} />
        </nav>
      </div>
      <div>
        <span className={styles.top__title}>Разделы</span>
        <nav className={styles.top__nav}>
          <FooterList items={topRightLinks} />
          <FooterCert />
        </nav>
      </div>
      <div>
        <span className={styles.top__title}>Служба поддержки</span>
        <div className={styles.top__wrapper}>
          <p>
            Мы всегда готовы вам помочь.
            <br /> Наши операторы онлайн 24/7
          </p>
          <FooterButtons />
          <FooterAsk />
        </div>
      </div>
      <div>
        <FooterWidget />
      </div>
    </div>
  );
};

export default FooterTop;
