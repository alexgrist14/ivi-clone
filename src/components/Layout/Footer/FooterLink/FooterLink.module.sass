@use '/src/styles/vars'

.link
  display: block
  color: vars.$primaryColor
  transition-duration: .4s
  transition-property: color,box-shadow,opacity

  &:hover, &:focus-visible
    color: vars.$whiteTextColor
