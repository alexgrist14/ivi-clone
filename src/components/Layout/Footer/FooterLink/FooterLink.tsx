import Link from "next/link";
import { FC } from "react";
import { useTranslation } from "next-i18next";
import styles from "./FooterLink.module.sass";
import ILink from "/src/types/ILink";

const FooterLink: FC<ILink> = ({ content, url, target }) => {
  const { t } = useTranslation("footer");

  return (
    <Link target={target} className={styles.link} href={url}>
      {t(content as string)}
    </Link>
  );
};

export default FooterLink;
