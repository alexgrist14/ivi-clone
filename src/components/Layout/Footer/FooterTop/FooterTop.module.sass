@use "src/styles/vars"

.top
  display: grid
  grid-template-columns: 1fr 1fr 1fr 1fr
  gap: 15px
  padding-block: 48px
  border-block: 1px solid rgba(255,255,255,.16)

  &__title
    display: block
    color: vars.$whiteTextColor
    margin-bottom: 8px
    font-weight: 700
    font-size: 15px
    line-height: 20px

  &__wrapper
    display: grid
    gap: 20px
    white-space: pre-line

  &__cert
    background-clip: text
    background-image: linear-gradient(-45deg,#c447ff,#ea003d)
    -webkit-text-fill-color: transparent
    margin-top: 8px

    & > a
      color: transparent
      font-size: 15px
      line-height: 20px
      font-weight: 500
