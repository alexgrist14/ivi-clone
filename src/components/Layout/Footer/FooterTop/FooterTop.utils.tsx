import ILink from "/src/types/ILink";

export const topLeftLinks: Array<ILink> = [
  {
    content: "about.links.0",
    url: "https://corp.ivi.ru/?_gl=1*15hvcv0*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTI5OTQ0Ny42MC4wLjA.",
    target: "_blank",
  },
  {
    content: "about.links.1",
    url: "https://corp.ivi.ru/career/?_gl=1*2qtg9j*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTMwMTQwMy41Ny4wLjA.#career-vacancy-block",
    target: "_blank",
  },
  {
    content: "about.links.2",
    url: "https://www.ivi.ru/pages/beta",
    target: "_blank",
  },
  {
    content: "about.links.3",
    url: "https://www.ivi.ru/info/partners",
    target: "_blank",
  },
  {
    content: "about.links.4",
    url: "https://corp.ivi.ru/advertisers/?_gl=1*uhgz7x*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTMwMTQwMy41Ny4wLjA.",
    target: "_blank",
  },
  {
    content: "about.links.5",
    url: "https://www.ivi.ru/info/agreement",
    target: "_blank",
  },
  {
    content: "about.links.6",
    url: "https://www.ivi.ru/info/confidential",
    target: "_blank",
  },
  {
    content: "about.links.7",
    url: "https://www.ivi.ru/info/goryachaya-liniya-komplaens",
    target: "_blank",
  },
];

export const topRightLinks: Array<ILink> = [
  {
    content: "sections.links.0",
    url: "https://www.ivi.ru/",
    target: "_blank",
  },
  {
    content: "sections.links.1",
    url: "https://www.ivi.ru/new",
    target: "_blank",
  },
  { content: "sections.links.2", url: "/movies" },
  {
    content: "sections.links.3",
    url: "https://www.ivi.ru/series",
    target: "_blank",
  },
  {
    content: "sections.links.4",
    url: "https://www.ivi.ru/animation",
    target: "_blank",
  },
  {
    content: "sections.links.5",
    url: "https://www.ivi.ru/tvplus",
    target: "_blank",
  },
  {
    content: "sections.links.6",
    url: "https://www.ivi.ru/goodmovies",
    target: "_blank",
  },
];
