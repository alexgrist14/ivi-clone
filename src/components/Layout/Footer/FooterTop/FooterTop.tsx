import { FC } from "react";
import styles from "./FooterTop.module.sass";
import FooterList from "../FooterList/FooterList";
import FooterButtons from "../FooterButtons/FooterButtons";
import FooterAsk from "../FooterAsk/FooterAsk";
import FooterWidget from "../FooterWidget/FooterWidget";
import { topLeftLinks, topRightLinks } from "./FooterTop.utils";
import FooterCert from "../FooterCert/FooterCert";
import { useTranslation } from "next-i18next";

const FooterTop: FC = () => {
  const { t } = useTranslation("footer");

  return (
    <div className={styles.top}>
      <div>
        <span className={styles.top__title}>{t("about.title")}</span>
        <nav className={styles.top__nav}>
          <FooterList items={topLeftLinks} />
        </nav>
      </div>
      <div>
        <span className={styles.top__title}>{t("sections.title")}</span>
        <nav className={styles.top__nav}>
          <FooterList items={topRightLinks} />
          <FooterCert />
        </nav>
      </div>
      <div>
        <span className={styles.top__title}>{t("support.title")}</span>
        <div className={styles.top__wrapper}>
          <p>{t("support.text")}</p>
          <FooterButtons />
          <FooterAsk />
        </div>
      </div>
      <div>
        <FooterWidget />
      </div>
    </div>
  );
};

export default FooterTop;
