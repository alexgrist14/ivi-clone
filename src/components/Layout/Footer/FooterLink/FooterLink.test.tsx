import { renderWithProviders } from "/src/utils/test-utils";
import { topLeftLinks } from "../FooterTop/FooterTop.utils";
import FooterLink from "./FooterLink";

describe("Footer link", () => {
  test("should renders without errors", () => {
    const render = renderWithProviders(
      <FooterLink {...topLeftLinks[0]} />
    );
    expect(render.component.container).toMatchSnapshot();
  });
});
