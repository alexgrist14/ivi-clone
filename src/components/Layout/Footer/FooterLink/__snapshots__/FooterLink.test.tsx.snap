// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer link should renders without errors 1`] = `
<div>
  <a
    class="link"
    href="https://corp.ivi.ru/?_gl=1*15hvcv0*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTI5OTQ0Ny42MC4wLjA."
    target="_blank"
  >
    about.links.0
  </a>
</div>
`;
